common/src/test/java/com/microsoft/identity/common/internal/providers/microsoft/nativeauth/integration/scenario/ResetPasswordScenarioTest.kt

```diff
@@ -156,11 +160,13 @@ class ResetPasswordScenarioTest {
         val mockResetPasswordSubmitCodeCommandParameters = mockk<ResetPasswordSubmitCodeCommandParameters>()
         every { mockResetPasswordSubmitCodeCommandParameters.getContinuationToken() } returns continuationToken
         every { mockResetPasswordSubmitCodeCommandParameters.getCode() } returns oobCode
+        every { mockResetPasswordSubmitCodeCommandParameters.getCorrelationId() } returns correlationId
         val ssprContinueResult = nativeAuthOAuth2Strategy.performResetPasswordContinue(
             mockResetPasswordSubmitCodeCommandParameters
         )
         Assert.assertTrue(ssprContinueResult is ResetPasswordContinueApiResult.PasswordRequired)
         continuationToken = (ssprContinueResult as ResetPasswordContinueApiResult.PasswordRequired).continuationToken
+        correlationId = ssprStartResult.correlationId ?: UUID.randomUUID().toString()
```

Comment:
why are we generating a new correlation id before each request?

Reply:
we're not, only if it's null. Do you think we shouldn't?
@diegojerezba  how are you doing this on iOS?

Reply:
When creating requests: 

On iOS we have nullable correlationIds only in the public interface. If the users don't set them, we generate them.
After that, our correlationId fields that we pass through functions cannot be null.

When decoding the correlationId from the headers:

Our API models allow for nullable correlationIds. If the correlationId is null, we reuse the same correlationId that was sent during the request (ie: the correlationId that was set at the public interface, either by the user or generated by us)

Reply:
> Our API models allow for nullable correlationIds. If the correlationId is null, we reuse the same correlationId that was sent during the request (ie: the correlationId that was set in the public interface, either by the user or generated by us)

this is a good point, which we're not doing right now. if the API returns null, it stays null. I'll fix that.